src/krb5_plugin/passkey/passkey_kdcpreauth.c

```diff
@@ -0,0 +1,429 @@
+                      krb5_data *req_pkt,
+                      krb5_kdc_req *request,
+                      krb5_enc_tkt_part *enc_tkt_reply,
+                      krb5_pa_data *pa,
+                      krb5_kdcpreauth_callbacks cb,
+                      krb5_kdcpreauth_rock rock,
+                      krb5_kdcpreauth_moddata moddata,
+                      krb5_kdcpreauth_verify_respond_fn respond,
+                      void *arg)
+{
+    struct sss_radiuskdc_state *state;
+    struct sss_passkeykdc_config *config = NULL;
+    struct sss_passkey_message *message = NULL;
+    char *configstr = NULL;
+    krb5_error_code ret;
+    krb5_data rstate;
+
+    state = (struct sss_radiuskdc_state *)moddata;
+
+    ret = sss_radiuskdc_enabled(SSSD_PASSKEY_CONFIG, kctx, cb, rock, &configstr);
+    if (ret != 0) {
+        goto done;
+    }
+
+    ret = sss_passkeykdc_config_init(state, kctx, cb->client_name(kctx, rock),
+                                     configstr, &config);
+    if (ret != 0) {
+        goto done;
+    }
+
+    ret = sss_radiuskdc_get_cookie(kctx, cb, rock, SSSD_PASSKEY_PADATA,
+                                   &rstate);
+    if (ret != 0) {
+        goto done;
+    }
+
+    if (pa->pa_type != SSSD_PASSKEY_PADATA || pa->length != 0) {
+        ret = KRB5_PREAUTH_BAD_TYPE;
+        goto done;
+    }
+
+    message = sss_passkey_message_decode_padata(pa);
+    if (message == NULL) {
+        ret = ENOMEM;
+        goto done;
+    }
+
+    if (message->phase != SSS_PASSKEY_PHASE_REPLY
+        || strlen(message->state) != rstate.length
+        || strncmp(message->state, rstate.data, rstate.length) != 0) {
+        ret = EINVAL;
+        goto done;
+    }
+
+    ret = sss_passkeykdc_verify_send(kctx, rock, cb, enc_tkt_reply, respond,
+            arg, message, (const char * const *)config->passkey->indicators,
+            config->radius);
+    if (ret != 0) {
+        goto done;
+    }
+
+    ret = 0;
+
+done:
+    if (ret != 0) {
+        respond(arg, ret, NULL, NULL, NULL);
+    }
+
+    cb->free_string(kctx, rock, configstr);
+    sss_passkeykdc_config_free(config);
+    sss_passkey_message_free(message);
+}
+
+krb5_error_code
+kdcpreauth_passkey_initvt(krb5_context kctx,
+                          int maj_ver,
+                          int min_ver,
+                          krb5_plugin_vtable vtable)
+{
+    static krb5_preauthtype pa_type_list[] = { SSSD_PASSKEY_PADATA, 0 };
+    krb5_kdcpreauth_vtable vt;
+
+    if (maj_ver != 1) {
+        return KRB5_PLUGIN_VER_NOTSUPP;
+    }
+
+    vt = (krb5_kdcpreauth_vtable)vtable;
+    vt->name = discard_const(SSSD_PASSKEY_PLUGIN);
+    vt->pa_type_list = pa_type_list;
+    vt->init = sss_passkeykdc_init;
+    vt->fini = sss_radiuskdc_fini;
+    vt->flags = sss_radiuskdc_flags;
+    vt->edata = sss_passkeykdc_edata;
+    vt->verify = sss_passkeykdc_verify;
+    vt->return_padata = sss_radiuskdc_return_padata;
+
+    com_err(SSSD_PASSKEY_PLUGIN, 0, "Loaded");
```

Comment:
Hi,

there are now three modules (certauth, IdP and passkey pre-auth) which just print `Loaded` in the logs:
```
Jan 02 12:38:47 master.ipa.vm krb5kdc[479561](info): Loaded
Jan 02 12:38:47 master.ipa.vm krb5kdc[479561](info): Loaded
Jan 02 12:38:47 master.ipa.vm krb5kdc[479561](info): Loaded
```
I think it would help to add the module's name here (and for the other modules) as well.

bye,
Sumit

Reply:
Changed for IdP and passkey. I was not able to find other "Loaded" string in SSSD nor in IPA.